contracts/savings/SavingsManager.sol

```diff
@@ -310,25 +414,32 @@ contract SavingsManager is ISavingsManager, PausableModule {
         }
     }
 
+
     /***************************************
-                MANAGEMENT
+            Revenue Redistribution
     ****************************************/
 
     /**
-     * @dev Withdraws any unallocated interest, i.e. that which has been saved for use
-     *      elsewhere in the system, based on the savingsRate
-     * @param _mAsset       mAsset to collect from
-     * @param _recipient    Address of mAsset recipient
+     * @dev Redistributes the unallocated interest to the saved recipient, allowing
+     * the siphoned assets to be used elsewhere in the system
+     * @param _mAsset  mAsset to collect
      */
-    function withdrawUnallocatedInterest(address _mAsset, address _recipient)
+    function distributeUnallocatedInterest(address _mAsset)
         external
         onlyGovernance
     {
+        IRevenueRecipient recipient = revenueRecipients[_mAsset];
+        require(address(recipient) != address(0), "Must have valid recipient");
+
         IERC20 mAsset = IERC20(_mAsset);
         uint256 balance = mAsset.balanceOf(address(this));
+        uint256 leftover = _allUnclaimedRewards(_mAsset);
+
+        uint256 unallocated = balance.sub(leftover);
 
-        emit InterestWithdrawnByGovernor(_mAsset, _recipient, balance);
+        mAsset.approve(address(recipient), unallocated);
+        recipient.depositFunds(_mAsset, unallocated);
```

Comment:
This isn't really depositing the funds per se. It's essentially notifying the recipient that it was approved, right?

Reply:
Yes - I should rename this to 'notifyRedistributionAmount'